src/main/java/frc/robot/RobotContainer.java

```diff
@@ -16,45 +36,74 @@
+    public RobotContainer() {
+      drivebase.setDefaultCommand(new curveDrive(drivebase, () -> -controller.getRawAxis(ControllerConstants.SPEED_JOYSTICK), () -> controller.getRawAxis(ControllerConstants.ROTATION_JOYSTICK), () -> controller.getRawButton(ControllerConstants.ROTATE_IN_PLACE_BUTTON), () -> controller.getRawAxis(ControllerConstants.THROTTLE_TRIGGER)));
 
-  }
+      configureButtonBindings();
+    }
 
-  public Command getAutonomousCommand() {
-    return null;
-  }
+    private void configureButtonBindings() {
 
-  public void setRumble(double rumble) {
-    controller.setRumble(RumbleType.kLeftRumble, rumble);
-    controller.setRumble(RumbleType.kRightRumble, rumble);
-  }
+    }
+
+    public Command generatePath(String pathName) {
+      try {
+        Path trajectoryPath = Filesystem.getDeployDirectory().toPath().resolve(pathName);
+        Trajectory trajectory = TrajectoryUtil.fromPathweaverJson(trajectoryPath);
+        RamseteCommand ramseteCommand = new RamseteCommand(
+          trajectory,
+          drivebase::getPose,
+          new RamseteController(AutoConstants.kRamseteB, AutoConstants.kRamseteZeta),
+          new SimpleMotorFeedforward(
+            DrivebaseConstants.ksVolts,
+            DrivebaseConstants.kvVoltSecondsPerMeter,
+            DrivebaseConstants.kaVoltSecondsSquaredPerMeter),
+          DrivebaseConstants.kDriveKinematics,
+          drivebase::getWheelSpeeds,
+          new PIDController(DrivebaseConstants.kPDriveVel, 0, 0),
+          new PIDController(DrivebaseConstants.kPDriveVel, 0, 0),
+          // RamseteCommand passes volts to the callback
+          drivebase::tankDriveVolts,
+          drivebase);
+        // Create and push Field2d to SmartDashboard.
+        Field2d m_field = new Field2d();
+        SmartDashboard.putData(m_field);
+
+        // Push the trajectory to Field2d.
+        m_field.getObject("traj").setTrajectory(trajectory);
 
-  public double applyDeadband(double value) {
-    double deadband = ControllerConstants.JOYSTICK_DEADBAND;
+        // Reset odometry to the starting pose of the trajectory.
+        drivebase.resetOdometry(trajectory.getInitialPose());
+        drivebase.putTrajectory(trajectory);
+        // Set up a sequence of commands
+        // First, we want to reset the drivetrain odometry
+        return new InstantCommand(() -> drivebase.resetOdometry(trajectory.getInitialPose()), drivebase)
+            // next, we run the actual ramsete command
+            .andThen(ramseteCommand)
+            // Finally, we make sure that the robot stops
+            .andThen(new InstantCommand(() -> drivebase.tankDriveVolts(0, 0), drivebase));
+      } catch (IOException ex) {
+        DriverStation.reportError("Unable to open trajectory: " + pathName, ex.getStackTrace());
+      }
+      return null;
+    }
+
+    public Command getAutonomousCommand() {
+      return new SequentialCommandGroup(
+        generatePath("paths/Forward.wpilib.json"),
```

Comment:
Won't this NullPointer if the path is not found? This error should be handled gracefully

Reply:
there's a try, catch. Is there anything else we can do to make it better?